crates/core/src/network/handler/mod.rs

```diff
@@ -488,22 +488,49 @@ fn get_meta_aspects_from_dht_eav(
                     Ok(EntryAspect::LinkAdd(link_data.clone(), header))
                 }
                 Attribute::RemovedLink(_, _) => {
-                    let (link_data, removed_link_entries) =
-                        unwrap_to!(value_entry.entry_with_meta.entry => Entry::LinkRemove);
-                    Ok(EntryAspect::LinkRemove(
-                        (link_data.clone(), removed_link_entries.clone()),
-                        header,
-                    ))
+                    match value_entry.entry_with_meta.entry {
+                        Entry::LinkRemove((link_data, removed_link_entries)) => {
+                            Ok(EntryAspect::LinkRemove(
+                                (link_data.clone(), removed_link_entries.clone()),
+                                header,
+                            ))
+                        },
+                        Entry::LinkAdd(link_data) => {
+                            //bad = true;
+//                            debug!("didn't get an expected LinkRemove, got: {:?}",value_entry.clone());
```

Comment:
We have to do a get_links here using entry adress, link tag, link type and then match on the target to get this array of values

Reply:
We can't do this.  It doesn't make sense to ask the network for the links when the network is asking us for data, remember this is part of a handle_fetch_entry call